### PR TITLE
Add check to the test bootstrapping for the maximum function level nesting

### PR DESCRIPTION
**Pull request by FGrosse**
I was just trying to run the whole test suit and ran into this issue.
When you have the xdebug extension loaded it will have a default maximum function level nesting of 100.
This is exceeded in the `NumberTheoryTests` so the tests for that fail.

It can be fixed easily by setting the php ini setting `xdebug.max_nesting_level = 150` (I just guessed 150 would be enough and it worked)

However it thought it might be a good idea to catch this issue before the tests run and notify the user about his bad configuration but still run the actual tests. This PR introduces this change

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -15,3 +15,27 @@ function buildPath()
 }
 
 include buildPath(__DIR__, '..', 'vendor', 'autoload.php');
+
+$requiredNestingLevel = 150;
+if (extension_loaded('xdebug')) {
```
Comment (aztech-dev) [inline tests/bootstrap.php]: An extra line here wouldn't hurt...

File: tests/bootstrap.php [hunk 2]
```diff
@@ -15,3 +15,27 @@ function buildPath()
...
+
+The tests will now run with this setting.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+TEXT
+        );
+        ini_set('xdebug.max_nesting_level', $requiredNestingLevel);
+    }
+}
```
Comment (aztech-dev) [inline tests/bootstrap.php]: Missing empty line at EOF

File: tests/bootstrap.php [hunk 3]
```diff
@@ -15,3 +15,27 @@ function buildPath()
 }
 
 include buildPath(__DIR__, '..', 'vendor', 'autoload.php');
+
+$requiredNestingLevel = 150;
+if (extension_loaded('xdebug')) {
+    $currentNestingLevel = intval(ini_get('xdebug.max_nesting_level'));
+    if ($currentNestingLevel < $requiredNestingLevel) {
```
Comment (aztech-dev) [inline tests/bootstrap.php]: Same here...
